packages/wallets/src/hooks/useInputDecimalFormatter.ts

```diff
@@ -0,0 +1,72 @@
+import { useCallback, useEffect, useState } from 'react';
+
+type TOptions = {
+    fraction_digits?: number;
+    with_sign?: boolean;
+};
+
+const useInputDecimalFormatter = (initial?: number, options?: TOptions) => {
```

Comment:
I see a lot of manual formatting. Is it possible to use: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat ? You can do conditions for the sign and let NumberFormat handle the rest

Reply:
This is the hooks that i copied from hooks package. This hook also doing input validation besides formatting, i will come back to it later to combine it with `NumberFormat`